### ISSUE TITLE
ISSN 2468-5542 | Journal of Islamic ethics (Brill) | Personenname mit Originalschrift

### ISSUE THREAD

**Issue by IxKo** [ubtue/DatenProbleme#1613]
**URL**
IxTheo#2021-11-09#2551D9FE6974226D7C05A4CA824A583AAB8C9F2E
https://brill.com/view/journals/jie/5/1-2/article-p331_11.xml

**Ausführliche Problembeschreibung**
Der Personenname ist falsch übertragen worden.
Solche Fälle, sollten entweder Translator-seitig korrigiert werden oder in den Errors landen.

![image](https://user-images.githubusercontent.com/57000004/141456466-b5db7e56-4115-47df-8a19-acc6204c4014.png)

**Comment by socheres** [comment]
Im RIS kommt der Datensatz wie folgt mit:

```
TY  - JOUR
TI  - المفاهيم الأخلاقية بين الائتمانية والعلمانية, written by طه عبد الرحمن
JO  - Journal of Islamic Ethics
AU  - هداري (Abdellah Haddari), عبدالله
AU  - هداري, عبدالله
AU  - , 
PY  - 2021
DA  - 20 Apr. 2021
PB  - Brill
CY  - Leiden, Niederlande
UR  - https://brill.com/view/journals/jie/5/1-2/article-p331_11.xml
DO  - https://doi.org/10.1163/24685542-12340062
VL  - 5
IS  - 1-2
SP  - 331
EP  - 342
LA  - English
ER  - 
```
Das Issue kann nicht Translator-seitig gelöst werden, daher ist es besser, ihn über QA-Regex wie im Fall https://github.com/ubtue/DatenProbleme/issues/1574#issuecomment-909340696 zu lösen.
@jriedl Kannst du bitte den regex für arabisch anpassen?

**Comment by jriedl** [comment]
`^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+)|(.*[\p{Han}\p{Arabic}\s,]+.*))$)`

Die Zeitschrift ist ja bereits live. Habe es jetzt testweise auf ub28 für `100` und `700`eingetragen.

**Comment by IxKo** [comment]
Da die Zs schon live ist, muss wohl auf das nächste Heft gewartet werden.

**Comment by IxKo** [comment]
Es scheint auf Nu zu funktionieren, denn ein Datensatz ist nun in der Error-Datei gelandet:

> Validating record IxTheo#2022-06-14#DBFEA8C5D51EE7090510E8BFEC0E3BF6C8C90E12...
Record IxTheo#2022-06-14#DBFEA8C5D51EE7090510E8BFEC0E3BF6C8C90E12 is invalid:
100: contents of subfield $a(Fan 林凡, Lin) did not match regex "^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+)|(.*\p{Han}.*))$)"
record with URL 'https://doi.org/10.1080/02549948.2021.1989780' already in database with the same hash (DBFEA8C5D51EE7090510E8BFEC0E3BF6C8C90E12)


Im Live-Betrieb ist ein solcher Fall bisher nicht mehr aufgetaucht. Da es aber für chines. Schriftzeichen {Han} funktioniert, sollte die RegEx zusätzlich mit {Arab} wahrscheinlich auch funktionieren und könnte global eingestellt werden.
Allerdings frage ich mich, ob nicht ein umgekehrter Fall eingerichtet werden könnte, der nicht ausschließt, sondern nur lateinische Schriftzeichen erlaubt?

**Comment by jriedl** [comment]
Ist jetzt auf nu und ub28. Da das Auftreten ja offenbar ohnehin extrem selten ist, wäre der Vorschlag, hier bei der vorhandenen Lösung zu bleiben.

**Comment by IxKo** [comment]
Ok, danke.

**Comment by IxKo** [comment]
Leider scheinen nun auch "normale" Namen ausgeschlossen zu werden:



Validating record IxTheo#2022-10-24#441644BDCCFDA291BA7F8776A7043ADB1075D8A8...
Record IxTheo#2022-10-24#441644BDCCFDA291BA7F8776A7043ADB1075D8A8 is invalid:
100: contents of subfield $a(**Poya, Abbas**) did not match regex "^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+)|(.*[\p{Han}\p{Arabic}\s,]+.*))$)"
700: contents of subfield $a(**Schatzschneider, Isabel**) did not match regex "^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+)|(.*[\p{Han}\p{Arabic}\s,]+.*))$)"
record with URL 'https://brill.com/view/journals/jie/6/1/article-p1_1.xml' already in database with the same hash (441644BDCCFDA291BA7F8776A7043ADB1075D8A8)


dafür werden die anderen Artikel wie gewünscht ausgefiltert.:

**Comment by jriedl** [comment]
Es ist für diese Zeitschrift  nun `^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+))$)([\p{Latin},\s]+)` auf ub28 für 100$a und 700$a hinterlegt.

**Comment by IxKo** [comment]
Danke, die übrigen Datensätze wurden nun eingespielt.
Könnte die neue RegEx nicht sogar für jede ZS eingestellt werden, da so auch andere Schriftzeichen wie chinesisch ausgeschlossen werden?

**Comment by jriedl** [comment]
> Als globale QA-Setting für Marc 100/700 ist bereits eine RegEx eingerichtet:

Ja, natürlich. Das Problem im konkreten Fall war ja, dass bei  `\p{Arabic}` die lateinischen Zeichen (warum auch immer) mitgematched haben und Stand jetzt löst das für obige Zeitschrift das Problem, wir haben aber nun nicht mehr, wie ursprünglich mal angedacht und favorisiert, einen Ausdruck mit einem reinen negativen Lookahead. 

Die eigentliche Grundfrage ist aber doch, wie oft ist der Fall einer fehlerhaften Durchleitung von Non-Latin-Zeichen in den vergangenen Monaten und Jahren für Zeitschriften, die sich auf den globalen Regex verlassen, tatsächlich aufgetreten ist. Selbst beim "Journal of Islamic Ethics" lag doch jetzt fast ein halbes Jahr dazwischen, bis überhaupt wieder Records mit dieser Eigenschaft aufgetaucht sind und ab einem bestimmten Punkt ist es doch schlicht eine Aufwand/Nutzen/Risiko-Abwägung.

**Comment by IxKo** [comment]
Die ZS "Journal of Islamic Ethics" erscheint nur einmal jährlich, daher tritt es momentan noch nicht häufig auf. Wenn mehr solcher ZSen ins zota-Verfahren überführt werden sollten, würde es sich sehr wahrscheinlich häufen.

Ich habe nun bei der ZS "Journal of Asian and African studies" auch die RegEx eingetragen, da hier (IxTheo#2022-10-07#CC540922AE795DFADE34F80D7F0D52F28CAB3875) bei einer Person chinesische Zeichen eingespielt worden sind.

**Comment by IxKo** [comment]
Das Problem scheint doch nicht gelöst. Heute wurde zwei Datensätze mit arabischen Zeichen in den Personennamen eingespielt:

IxTheo#2022-11-22#82D74AFEFD1740E053082B5CCC2332A49321F74A
IxTheo#2022-11-22#837C46FD58ED115A591D46A222F46B71C2DC6027

![image](https://user-images.githubusercontent.com/57000004/203528694-6aded722-da67-4e61-834d-686d81827b3f.png)

**Comment by IxKo** [comment]
Problem scheint behoben, danke.
